tck/methods/account.go

```diff
@@ -0,0 +1,234 @@
+		Balance:                       resp.Balance.String(),
+		Key:                           resp.Key.String(),
+		AccountMemo:                   resp.AccountMemo,
+		MaxAutomaticTokenAssociations: resp.MaxAutomaticTokenAssociations,
+		AutoRenewPeriod:               resp.AutoRenewPeriod}, nil
+}
+
+// UpdateAccount transaction to be associated with hedera.AccountUpdateTransaction and to be enough generic, so different
+// updates can be calling this func
+func (a *AccountService) UpdateAccount(_ context.Context, params param.UpdateAccountParams) (*response.AccountResponse, error) {
+	if params.NewPrivateKey != "" { // Proceed with UpdateAccountMemo
+		return a.UpdateAccountKey(params)
+	} else if params.Memo != "" { // Proceed with UpdateAccountKey
+		return a.UpdateAccountMemo(params)
+	}
+	return nil, nil
+}
+
+// DeleteAccount deletes a provided account by signing the transaction with the key of that account
+func (a *AccountService) DeleteAccount(_ context.Context, param param.DeleteAccountParams) (*response.AccountResponse, error) {
+	accId, _ := hedera.AccountIDFromString(param.AccountId)
+	recipientId, _ := hedera.AccountIDFromString(param.RecipientId)
+	tx, err := hedera.NewAccountDeleteTransaction().SetGrpcDeadline(&threeSecondsDuration).SetAccountID(accId).SetTransferAccountID(recipientId).FreezeWith(a.sdkService.Client)
+	if err != nil {
+		return nil, err
+	}
+	signature, _ := hedera.PrivateKeyFromString(param.AccountKey)
+
+	resp, err := tx.Sign(signature).Execute(a.sdkService.Client)
+	if err != nil {
+		return nil, err
+	}
+
+	receipt, _ := resp.GetReceipt(a.sdkService.Client)
+	return &response.AccountResponse{Status: receipt.Status.String()}, nil
+}
+
+// UpdateAccountKey updates an existing acoount id keys with provided params
+func (a *AccountService) UpdateAccountKey(params param.UpdateAccountParams) (*response.AccountResponse, error) {
+	accId, err := hedera.AccountIDFromString(params.AccountId)
+	if err != nil {
+		return nil, err
+	}
+	key, err := hedera.PublicKeyFromString(params.NewPublicKey)
+	if err != nil {
+		return nil, err
+	}
+
+	tx, err := hedera.NewAccountUpdateTransaction().SetGrpcDeadline(&threeSecondsDuration).SetAccountID(accId).SetKey(key).FreezeWith(a.sdkService.Client)
+	if err != nil {
+		return nil, err
+	}
+
+	oldKey, err := hedera.PrivateKeyFromString(params.OldPrivateKey)
+	if err != nil {
+		return nil, err
+	}
+	newKey, err := hedera.PrivateKeyFromString(params.NewPrivateKey)
+	if err != nil {
+		return nil, err
+	}
+	tx.Sign(oldKey)
+	tx.Sign(newKey)
+
+	resp, err := tx.Execute(a.sdkService.Client)
+	if err != nil {
+		return nil, err
+	}
+
+	receipt, err := resp.GetReceipt(a.sdkService.Client)
+	if err != nil {
+		return nil, err
+	}
+
+	return &response.AccountResponse{Status: receipt.Status.String()}, nil
+}
+
+// UpdateAccountMemo updates account memo of an existing account ID
+func (a *AccountService) UpdateAccountMemo(params param.UpdateAccountParams) (*response.AccountResponse, error) {
```

Comment:
make private

Reply:
Not sure it should be left private since the TCK has a specified test for `UpdateAccountMemo`. Not fully sure tho.

Reply:
Yes, you are right. We should add these methods to the assigner map.